src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentService.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// Empty interface just to mark document services.
+    /// </summary>
+    internal interface IDocumentService
```

Comment:
What is a 'DOcument Service'?  i.e. i understand waht an IWorkspaceService is, and how it allows different workspace layers to inject appropriate implementations of things.  I understand what an ILanguageService is, and how it allows language specific impls of behavior depending on the language of a doc.  But i have no idea hwat an IDocumentService is currently (even though this is at the bottom of this PR).

Reply:
> ...  how it allows different workspace layers to inject appropriate implementations of things

This is just a side effect of the implementation.

* IWorkspaceService is a service associated with a workspace
* ILanguageService is a service associated with a workspace and language
* IDocumentService is a service whose instances are associated with a document

Reply:
we want to show different presentation on FAR based on document. and we want that behavior to be injected by one that owns the document.

for example, for razor, we want them to inject document to us and anything around that document, we want them to own experience. for references on that document, they decide how to show it. this is first part. but we want to expent it to formatting, diagnostics and etc.

Reply:
whether they will share same service for multiple razor documents or not is implementation detail. we allow them to have different service per document or even a snapshot of a document. it is up to them.

Reply:
Right.. but my point is: this is all possible anyways with the existing system.  That said, we opened up an open design question about doing things this way.  So i'll wait for the conclusion of that meeting.